data/s3dis/indoor3d_util.py

```diff
@@ -0,0 +1,690 @@
+            assumes the data is shifted (min point is origin) and aligned
+            (aligned with XYZ axis)
+        label: N size uint8 numpy array from 0-12
+        num_point: int, how many points to sample in each block
+        block_size: float, physical size of the block in meters
+        stride: float, stride for block sweeping
+        random_sample: bool, if True, we will randomly sample block in the room
+        sample_num: int, if random sample, how many blocks to sample
+            [default: room area]
+        sample_aug: if random sample, how much aug
+    Returns:
+        block_datas: K x num_point x 6 np array of XYZRGB, RGB is in [0,1]
+        block_labels: K x num_point x 1 np array of uint8 labels
+
+    TODO: for this version, blocking is in fixed, non-overlapping pattern.
+    """
+    assert (stride <= block_size)
+
+    limit = np.amax(data, 0)[0:3]
+
+    # Get the corner location for our sampling blocks
+    xbeg_list = []
+    ybeg_list = []
+    if not random_sample:
+        num_block_x = int(np.ceil((limit[0] - block_size) / stride)) + 1
+        num_block_y = int(np.ceil((limit[1] - block_size) / stride)) + 1
+        for i in range(num_block_x):
+            for j in range(num_block_y):
+                xbeg_list.append(i * stride)
+                ybeg_list.append(j * stride)
+    else:
+        num_block_x = int(np.ceil(limit[0] / block_size))
+        num_block_y = int(np.ceil(limit[1] / block_size))
+        if sample_num is None:
+            sample_num = num_block_x * num_block_y * sample_aug
+        for _ in range(sample_num):
+            xbeg = np.random.uniform(-block_size, limit[0])
+            ybeg = np.random.uniform(-block_size, limit[1])
+            xbeg_list.append(xbeg)
+            ybeg_list.append(ybeg)
+
+    # Collect blocks
+    block_data_list = []
+    block_label_list = []
+    idx = 0
+    for idx in range(len(xbeg_list)):
+        xbeg = xbeg_list[idx]
+        ybeg = ybeg_list[idx]
+        xcond = (data[:, 0] <= xbeg + block_size) & (data[:, 0] >= xbeg)
+        ycond = (data[:, 1] <= ybeg + block_size) & (data[:, 1] >= ybeg)
+        cond = xcond & ycond
+        if np.sum(cond) < 100:  # discard block if there are less than 100 pts.
+            continue
+
+        block_data = data[cond, :]
+        block_label = label[cond]
+
+        # randomly subsample data
+        block_data_sampled, block_label_sampled = \
+            sample_data_label(block_data, block_label, num_point)
+        block_data_list.append(np.expand_dims(block_data_sampled, 0))
+        block_label_list.append(np.expand_dims(block_label_sampled, 0))
+
+    return np.concatenate(block_data_list, 0), \
+        np.concatenate(block_label_list, 0)
+
+
+def room2blocks_plus(data_label, num_point, block_size, stride, random_sample,
+                     sample_num, sample_aug):
+    """room2block with input filename and RGB preprocessing."""
```

Comment:
Is it a conventional function name in this field? I mean the "_plus". If not maybe we can make it more clear especially in terms of the difference with the room2blocks.